internal/jobs/client.go

```
@@ -0,0 +1,183 @@
+	}
+	// The lock lives on its own connection, not a pooled one: Migrate talks
+	// to the pool, and a one-connection pool would deadlock if the lock sat
+	// on that only slot.
+	lockConn, err := pgx.ConnectConfig(ctx, c.pool.Config().ConnConfig)
+	if err != nil {
+		return fmt.Errorf("failed to open a connection for the river migration lock: %w", err)
+	}
+	lockCtx, cancel := context.WithTimeout(ctx, 5*time.Minute)
+	if _, err := lockConn.Exec(lockCtx, "SELECT pg_advisory_lock($1)", riverMigrationLockID); err != nil {
+		cancel()
+		_ = lockConn.Close(context.Background())
```

golangci-lint (2.12.2)

[error] Non-inherited new context, use function like `context.WithXXX` instead

(contextcheck)